docs/production.md

```diff
@@ -0,0 +1,420 @@
+  --redelivery-timeout 10m \
+  --healthcheck-port 8080 \
+  --metrics-port 9090 \
+  --logging-format json \
+  --tasks myapp.tasks:production_tasks
+```
+
+### Tuning for Different Workloads
+
+**High-throughput, fast tasks:**
+```bash
+docket worker \
+  --concurrency 100 \
+  --redelivery-timeout 30s \
+  --minimum-check-interval 50ms \
+  --scheduling-resolution 100ms
+```
+
+**Long-running, resource-intensive tasks:**
+```bash
+docket worker \
+  --concurrency 5 \
+  --redelivery-timeout 1h \
+  --minimum-check-interval 1s \
+  --scheduling-resolution 5s
+```
+
+**Mixed workload with perpetual tasks:**
+```bash
+docket worker \
+  --concurrency 25 \
+  --redelivery-timeout 5m \
+  --schedule-automatic-tasks \
+  --tasks myapp.tasks:all_tasks,myapp.monitoring:health_checks
+```
+
+## Connection Management
+
+### Redis Connection Pools
+
+Docket automatically manages Redis connection pools, but you can tune them for your environment:
+
+```python
+from redis.asyncio import ConnectionPool
+
+# Custom connection pool for high-concurrency workers
+pool = ConnectionPool.from_url(
+    "redis://redis-cluster:6379/0",
+    max_connections=50,  # Match or exceed worker concurrency
+    retry_on_timeout=True,
+    socket_keepalive=True,
+    socket_keepalive_options={
+        1: 1,  # TCP_KEEPIDLE
+        2: 3,  # TCP_KEEPINTVL
+        3: 5,  # TCP_KEEPCNT
+    }
+)
+
+async with Docket(name="orders", connection_pool=pool) as docket:
+    # Use the custom pool
+    pass
+```
+
+### Redis Cluster and High Availability
+
+Docket works with Redis clusters and sentinel configurations:
+
+```python
+# Redis Cluster
+docket_url = "redis://redis-cluster:6379/0?cluster=true"
+
+# Redis Sentinel
+docket_url = "redis+sentinel://sentinel1:26379,sentinel2:26379/mymaster"
+
+# With authentication
+docket_url = "redis://:password@redis.prod.com:6379/0"
+```
+
+### Valkey Support
+
+Docket also works with Valkey (Redis fork):
+
+```bash
+export DOCKET_URL=valkey://valkey.prod.com:6379/0
+```
+
+## Monitoring and Observability
+
+### Prometheus Metrics
+
+Enable Prometheus metrics with the `--metrics-port` option:
+
+```bash
+docket worker --metrics-port 9090
+```
+
+Available metrics include:
+
+**Task Counters:**
+- `docket_tasks_added` - Tasks scheduled
+- `docket_tasks_started` - Tasks begun execution
+- `docket_tasks_succeeded` - Successfully completed tasks
+- `docket_tasks_failed` - Failed tasks
+- `docket_tasks_retried` - Retry attempts
+- `docket_tasks_stricken` - Tasks blocked by strikes
+
+**Task Timing:**
+- `docket_task_duration` - Histogram of task execution times
+- `docket_task_punctuality` - How close tasks run to their scheduled time
+
+**System Health:**
+- `docket_queue_depth` - Tasks ready for immediate execution
+- `docket_schedule_depth` - Tasks scheduled for future execution
+- `docket_tasks_running` - Currently executing tasks
+- `docket_redis_disruptions` - Redis connection failures
+- `docket_strikes_in_effect` - Active strike rules
+
```

Comment:
these got formatted oddly in the resulting page:

![image](https://github.com/user-attachments/assets/31991b5b-d02e-49e8-943c-b5b5089cbc54)